src/main/java/org/sonatype/nexus/ci/nxrm/v3/ComponentUploaderNxrm3.groovy

```diff
@@ -34,9 +37,21 @@ import static org.sonatype.nexus.ci.util.RepositoryManagerClientUtil.nexus3Clien
 class ComponentUploaderNxrm3
     extends ComponentUploader
 {
+  @Override
+  void maybeCreateTag(@Nullable final String tagName) {
+    if (tagName?.trim()) {
+      def nxrmClient = getRepositoryManagerClient(nxrmConfiguration)
+      Optional<Tag> existingTag = nxrmClient.getTag(tagName)
```

Comment:
trivial: could be trimmed to a 1-line if desired
`nxrmClient.getTag(tagName).orElse(nxrmClient.createTag(tagName))`